charts/generic/templates/deployment.yaml

```diff
@@ -22,8 +22,12 @@ spec:
   {{- end }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
-      annotations: {{- toYaml . | nindent 8 }}
+      {{- if or .Values.podAnnotations .Values.configMap.enabled }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```

Comment:
Shouldn't we exclude the annotation if the configMap is disabled?

Reply:
Yes. Yes, we should :D